InventoryService/cmd/controllers/item_controller.go

```diff
@@ -0,0 +1,263 @@
+
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusBadRequest, err.Error(), nil)
+		return
+	}
+
+	if filters.AlcoholMin < 0 || filters.AlcoholMax > 1 {
+		resources.ResponseJSON(c, http.StatusBadRequest, resources.InadequateAlcohol, nil)
+		return
+	}
+
+	ingridients, err := controller.Client.Ingridient.
+		Query().
+		Where(
+			ingridient.And(
+				ingridient.NameContainsFold(filters.Name),
+				ingridient.AlcoholGT(filters.AlcoholMin),
+				ingridient.AlcoholLT(filters.AlcoholMax),
+				ingridient.IsDry(filters.IsDry),
+			),
+		).
+		All(c)
+
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusInternalServerError, err.Error(), nil)
+		return
+	}
+
+	resources.ResponseJSON(c, http.StatusOK, "", ingridients)
+}
+
+// @BasePath /api/v1
+
+// AddIngridient godoc
+// @Summary           Adds new ingridient
+// @Description       Add new ingridient to database
+// @Tags              Inventory manipulation
+// @Accept            json
+// @Produce           application/json
+// @Param             item body models.ItemModelCreate true "Item data"
+// @Success           201
+// @Failure           400
+// @Router            /ingridient [post]
+func (controller *ItemController) AddIngridient(c *gin.Context) {
+	jsonData, err := c.GetRawData()
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusBadRequest, err.Error(), nil)
+		return
+	}
+
+	var item models.ItemModelCreate
+	json.Unmarshal(jsonData, &item)
+
+	passedValidation, errorMessage := controller.ValidateNewIngridient(c, item)
+
+	if !passedValidation {
+		resources.ResponseJSON(c, http.StatusBadRequest, errorMessage, nil)
+		return
+	}
+
+	ingridient, err := controller.Client.Ingridient.
+		Create().
+		SetName(item.Name).
+		SetAlcohol(item.Alcohol).
+		SetMeasurmentUnit(ingridient.MeasurmentUnit(item.MeasurmentUnit)).
+		SetIsDry(item.IsDry).
+		Save(c)
+
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusInternalServerError, err.Error(), nil)
+		return
+	}
+
+	resources.ResponseJSON(c, http.StatusCreated, resources.IngridientAdded, ingridient)
+}
+
+// @BasePath /api/v1
+
+// AddIngridient godoc
+// @Summary           Changes ingridient information
+// @Description       Change ingridient in the database by id
+// @Tags              Inventory manipulation
+// @Accept            json
+// @Produce           application/json
+// @Param             item body models.ItemModel true "Item id and it's new data"
+// @Success           202
+// @Failure           400
+// @Failure           404
+// @Router            /ingridient [put]
+func (controller *ItemController) ChangeIngridient(c *gin.Context) {
+	jsonData, err := c.GetRawData()
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusBadRequest, err.Error(), nil)
+		return
+	}
+
+	var item models.ItemModel
+	json.Unmarshal(jsonData, &item)
+
+	ingridientOld, err := controller.Client.Ingridient.
+		Query().
+		Where(ingridient.ID(item.Id)).
+		Only(c)
+
+	if err != nil {
+		if ingridientOld == nil {
```

Comment:
I think it's not possible with `Only(c)` call

Reply:
Why? It's clearly possible because it works

Reply:
Ah, sorry, I misunderstood the code. Nit: it's better to use `if ent.IsNotFound(err) {}` instead of `if ingridientOld == nil` here to improve code readability